src/python/pants/fs/archive.py

```diff
@@ -22,8 +22,28 @@
 class Archiver(AbstractClass):
 
   @classmethod
-  def extract(cls, path, outdir):
-    """Extracts an archive's contents to the specified outdir."""
+  def extract(cls, path, outdir, filter_func=None, safe=False):
+    """Extracts an archive's contents to the specified outdir with an optional filter.
+
+    :API: public
+
+    :param string path: path to the zipfile to extract from
+    :param string outdir: directory to extract files into
+    :param function filter_func: optional filter with the filename as the parameter.  Returns True
+      if the file should be extracted.  Note that filter_func is ignored for non-zip archives.
+    :param bool safe: True if use safe extraction.  Safe extraction ignores errors if destination
+      already exists and thus is safe to use in different threads / concurrent processes.
+    """
+    if safe:
+      with temporary_dir() as temp_dir:
```

Comment:
Should prefer https://github.com/pantsbuild/pants/blob/1889b938596746110696f9ff58c1985f76635b6a/src/python/pants/util/dirutil.py#L311-L319 for this usecase, which handles the temp directory creation as well.

Reply:
yes, much better.

Reply:
`uuid.uuid4().hex` will definitely not contain unicode... not sure what you mean there. But not a blocker.